src/main/java/org/wso2/carbon/inbound/kafka/KafkaMessageConsumer.java

```diff
@@ -130,25 +133,41 @@ private void commitRecords(ConsumerRecords<byte[], byte[]> records) {
                 // Fixes - CS0196510
                 Object value  = record.value();
                 boolean isConsumed = false;
+                boolean poisonPillDetected = false;
                 if ( value == null ) {
+                    if (isPoisonPill(record)) {
+                        poisonPillDetected = true;
+                        log.warn("A poison pill was detected and will be skipped. The respective error details"
```

Comment:
should we print the correlation id as well?

Reply:
There is no specific correlation id at this level. What we can print is the Topic, Key or Value, Offset, and partition kind of details if needed.

Reply:
https://github.com/wso2-extensions/esb-inbound-kafka/pull/59/commits/ff12250e6c49f14d72ef658730c55f49849ce8a4